tests/testthat/test-empirical.R

```diff
@@ -0,0 +1,190 @@
+# h_get_empirical ----
+
+test_that("h_get_empirical obtain empirical covariance", {
+  fit <- get_mmrm_emp()
+  expect_snapshot_tolerance(h_get_empirical(fit$tmb_data, fit$theta_est, fit$beta_est, fit$beta_vcov, FALSE))
+})
+
+test_that("h_get_empirical obtain jackknife covariance", {
+  fit <- get_mmrm_jack()
+  expect_snapshot_tolerance(h_get_empirical(fit$tmb_data, fit$theta_est, fit$beta_est, fit$beta_vcov, TRUE))
+})
+
+# integration test ----
+
+# unweighted mmrm ----
+
+test_that("empirical covariance are the same with SAS result for ar1", {
+  fit <- mmrm(
+    FEV1 ~ ARMCD + ar1(AVISIT | USUBJID),
+    data = fev_data,
+    cov = "Empirical", method = "Residual"
+  )
+  expected <- 0.82581291651503
+  expect_equal(sqrt(component(fit, "beta_vcov")[2, 2]), expected, tolerance = 1e-4)
```

Comment:
ok so here and below we just always look at a single diagonal variance element of beta_vcov. can we maybe look at the whole 2x2 matrix here? so all 4 elements. that would give more confidence, especially also for the correlation adjustment

Reply:
let me go through the SAS guide to understand how the covariance can be displayed (currently it is only available in LSMEANS for now)

Reply:
only updated the new added empirical and weighted empirical; Jackknife is not added because SAS do not have Jackknife; Other integration tests like Kenward-Roger and Satterthwaite are not updated as they are also checking the d.f. so still using the lsmeans result. I wonder if we should create a folder to hold these integration tests and save the results appropriately in `inst` folder, and compare the result with simple code, instead of copying the results from SAS log

Reply:
thanks Liming - yeah there is pros and cons. I would tend to prefer continuing having the numbers in the test directly